Bio/SeqAnalysis/SeqAnalyzer.py

```diff
@@ -100,27 +107,49 @@ def main_calculations(id_seq_dictionary: dict) -> tuple:
+        [{'Helix': 22.222, 'Turn': 16.667, 'Sheet': 22.222}, {'Helix': 21.053, 'Turn': 15.789, 'Sheet': 21.053}],
+        [-1.661, -1.758], [{'Cysteines_reduced': 6990, 'Cysteines_residues': 6990},
+        {'Cysteines_reduced': 6990, 'Cysteines_residues': 6990}])
         """
         list_of_proteinanalysis_objects = [ProteinAnalysis(seq) for seq in id_seq_dictionary.values()]
-        molecular_weight = [seq.molecular_weight() for seq in list_of_proteinanalysis_objects]
-        pi = [seq.isoelectric_point() for seq in list_of_proteinanalysis_objects]
+        molecular_weight = [round(seq.molecular_weight(), 3) for seq in list_of_proteinanalysis_objects]
+        pi = [round(seq.isoelectric_point(), 3) for seq in list_of_proteinanalysis_objects]
         amino_acids = [seq.count_amino_acids() for seq in list_of_proteinanalysis_objects]
-        aromaticity = [seq.aromaticity() for seq in list_of_proteinanalysis_objects]
-        amino_acids_percent = [seq.get_amino_acids_percent() for seq in list_of_proteinanalysis_objects]
-        instability = [seq.instability_index() for seq in list_of_proteinanalysis_objects]
-        flexibility = [sum(seq) / len(seq) for seq in [aa.flexibility() for aa in list_of_proteinanalysis_objects]]
-        secondary_structure_fraction = [seq.secondary_structure_fraction() for seq in
-                                        list_of_proteinanalysis_objects]
-        gravy = [seq.gravy() for seq in list_of_proteinanalysis_objects]
-        mol_ext_coefficient = [seq.molar_extinction_coefficient() for seq in list_of_proteinanalysis_objects]
+        aromaticity = [round(seq.aromaticity(), 3) for seq in list_of_proteinanalysis_objects]
+        amino_acids_percent = [dict(seq.get_amino_acids_percent()) for seq in list_of_proteinanalysis_objects]
+        for dict_aa_percent in amino_acids_percent:  # rounding values in aa percentage occurrences dictionaries
+            for key, value in dict_aa_percent.items():
+                dict_aa_percent[key] = round(value * 100, 3)
+
+        instability = [round(seq.instability_index(), 3) for seq in list_of_proteinanalysis_objects]
+        flexibility = [
+            round(sum(seq) / len(seq), 3) for seq in
+            [aa.flexibility() for aa in list_of_proteinanalysis_objects]
+                        ]
+        secondary_structure_fraction = [
+            {'Helix': round(v1 * 100, 3), 'Turn': round(v2 * 100, 3), 'Sheet': round(v3 * 100, 3)} for
+            v1, v2, v3 in (seq.secondary_structure_fraction() for seq in list_of_proteinanalysis_objects)
+                                        ]
+        gravy = [round(seq.gravy(), 3) for seq in list_of_proteinanalysis_objects]
+        mol_ext_coefficient = [
+            {'Cysteines_reduced': v1, 'Cysteines_residues': v2} for v1, v2 in
+            (seq.molar_extinction_coefficient() for seq in list_of_proteinanalysis_objects)
+                                 ]
```

Comment:
Again:

- rounding should be configurable
- complex comprehensions should be done more readable, i.e.

```
flexibility = [aa.flexibility() for aa in list_of_proteinanalysis_objects]
flexibility = [
    round(sum(flex) / len(flex), self.rounding)
    for flex in flexibility
]
```